codesearch/github/github.go

```diff
@@ -0,0 +1,73 @@
+// TODO(jdelfino): Move common github repo extraction code to this file from
+// cli.go and server.go
+package github
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/buildbuddy-io/buildbuddy/codesearch/schema"
+	"github.com/buildbuddy-io/buildbuddy/server/util/git"
+
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/go-enry/go-enry/v2"
+
+	xxhash "github.com/cespare/xxhash/v2"
+)
+
+const (
+	maxFileLen = 10_000_000
+
+	// The maximum amount of bytes from a file to use for language and
+	// mimetype detection.
+	detectionBufferSize = 1000
+)
+
+// TODO(tylerw): this should come from a flag?
+var (
+	skipMime = regexp.MustCompile(`^audio/.*|video/.*|image/.*|application/gzip$`)
+)
+
+func ExtractFields(name, commitSha string, repoURL *git.RepoURL, fileContent []byte) (map[string][]byte, error) {
```

Comment:
I always hate returning maps from a function, but because the field and value types are different you can't really return a slice of []byte slices or something. Nothing to be done here I guess.

Reply:
If it helps, when I move the rest of the github stuff into this package, this method can probably change to avoid the need to return this, and instead call MakeDocument directly on the DocumentSchema